### PR TITLE
users: Introduce group create dialog (Splitted out from https://github.com/cockpit-project/cockpit/pull/18136)

### PR DESCRIPTION
**Pull request by KKoukiou**
The user is required to specify the new group name and optionally they can also choose they group id.

### CODE REVIEW

File: pkg/users/dialog-utils.js [hunk 1]
```diff
@@ -64,3 +64,10 @@ function show_error_dialog(title, message) {
 export function show_unexpected_error(error) {
     show_error_dialog(_("Unexpected error"), error.message || error);
 }
+
+export function is_valid_char_name(c) {
+    return (c >= 'a' && c <= 'z') ||
+        (c >= 'A' && c <= 'Z') ||
```
Comment (cockpituous) [inline pkg/users/dialog-utils.js:70]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230131-133441-a322fc08-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+export function group_create_dialog(groups, setGroupsCardExpanded) {
+    //  Handle also the case where logindef == null, i.e. the file does not exist.
+    //  While that's unusual, "empty /etc" is a goal, and it shouldn't crash the page.
+    cockpit.file("/etc/login.defs").read()
+            .then(logindef => {
+                const gids = groups
+                        .filter(g => g.name !== 'nobody')
+                        .map(group => group.gid);
+                const min_gid = parseInt(logindef.match(/^GID_MIN\s+(\d+)/m)[1] || 500);
+                const max_gid = parseInt(logindef.match(/^GID_MAX\s+(\d+)/m)[1] || 60000);
```
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:97]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230131-133441-a322fc08-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+                change("id", (Math.max(min_gid, Math.max(...gids.filter(id => id < max_gid)) + 1) + 1).toString());
+            }, () => Promise.resolve());
```
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:100]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230131-133441-a322fc08-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/dialog-utils.js (hunk 1 above)
Comment (cockpituous) [inline pkg/users/dialog-utils.js:70]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230131-152924-31f827fb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js (hunk 2 above)
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:97]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230131-152924-31f827fb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js (hunk 3 above)
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:100]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230131-152924-31f827fb-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/dialog-utils.js (hunk 1 above)
Comment (cockpituous) [inline pkg/users/dialog-utils.js:70]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230201-111629-cb712343-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js (hunk 3 above)
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:100]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230201-111629-cb712343-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/dialog-utils.js (hunk 1 above)
Comment (cockpituous) [inline pkg/users/dialog-utils.js:70]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230201-135749-6248d46c-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js (hunk 2 above)
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:97]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230201-135749-6248d46c-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js (hunk 3 above)
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:100]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230201-135749-6248d46c-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/dialog-utils.js (hunk 1 above)
Comment (cockpituous) [inline pkg/users/dialog-utils.js:70]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230201-153652-aa111af0-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js (hunk 2 above)
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:97]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230201-153652-aa111af0-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/group-create-dialog.js (hunk 3 above)
Comment (cockpituous) [inline pkg/users/group-create-dialog.js:100]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18271-20230201-153652-aa111af0-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by martinpitt — APPROVED** [review]
I only did a code review here, not an UI review. That looks good to me, thank you!

**Pull request merged**